src/app/habits/habiticons.tsx

```diff
@@ -0,0 +1,76 @@
+import Icon from "@/src/components/Icon";
+import { Text, View } from "@/src/components/Themed";
+import DefaultColors from "@/src/constants/DefaultColors";
+import { IconChevronLeft } from "@tabler/icons-react-native";
+import { router } from "expo-router";
+import { useColorScheme } from "nativewind";
+import { useContext, useMemo } from "react";
+import { TouchableOpacity } from "react-native";
+import { IconContext } from "./_layout";
+import { IconButton } from "./icon-button";
+import { icons } from "./icons";
+
+type HabitIconsTuple = [string, string[]];
+
+export default function HabitIcons() {
+  const { colorScheme } = useColorScheme();
+  const borderColor = DefaultColors[colorScheme].tint;
+  const { icon, setIcon } = useContext(IconContext);
+
+  const habitIcons: HabitIconsTuple[] = useMemo(() => {
+    return [
+      ["Health & Fitness", ["gym", "yoga", "healthyEating"]],
+      ["Productivity & Work", ["work", "timeManagement", "scheduling"]],
+      ["Personal Development", ["learning", "ideas", "careerAdvancement"]],
+      ["Leisure & Recreation", ["painting", "music", "photography"]],
+      ["Social & Family", ["socializing", "love", "familyTime"]],
+      [
+        "Mindfullness & Spirituality",
+        ["meditation", "religion", "spirituality"],
+      ],
+      ["Financial", ["saving", "spending", "investing"]],
+    ];
+  }, []);
```

Comment:
I feel like having this and icons.tsx is overly complicated. You could just have a file in the constants folder that has an object like this:

```ts
export const habitIcons = {
    "Health & Fitness": [
        IconBarbell,
        IconYoga,
        IconLeaf,
    ],
    ...
}
```

And then you can iterate over Object.keys(habitIcons). For each one you show the title and then iterate over habitIcons[section] for the icons